### ISSUE TITLE
Have validate accept null and undefined

### ISSUE THREAD

**Issue by grosch** [uuidjs/uuid#606]
Just to make life easier, it would be great if when I call `uuidValidate` and pass in `null` or `undefined`, it returned false.  Right now the compiler doesn't allow a potentially null/undefined value to be given.

Right now I'm doing something like this:

```typescript
this.route.paramMap
    .pipe(
        map(x => x.get('id')),
        filter(x => !!x && uuidValidate(x)),
        mustHaveValue(),
        switchMap(x => this.clientService.get(x)),
        untilDestroyed(this)
    )
```

**Comment by ctavan** [comment]
I think it could make sense to fix that on

**Comment by LinusU** [comment]
If you are sending a PR, I would recommend to change it to from `(uuid: string) => boolean` to `(uuid: any) => uuid is string`. That way TypeScript will know that you can pass anything in to the `validate` function, and that if `true` was returned it was definitely a `string`.

**Comment by broofa** [comment]
@LinusU Can you provide a link to the typescript docs that discuss whatever that behavior is that you're describing?  I'm still learning TS and this sort of conditional typing sounds interesting.

**Comment by grosch** [comment]
> @LinusU Can you provide a link to the typescript

https://www.typescripttutorial.net/typescript-tutorial/typescript-type-guards/

**Comment by LinusU** [comment]
some more refs:

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates
https://stackoverflow.com/q/40081332/148072

**Comment by ctavan** [comment]
Reopening as this may be solved in #654.

**Comment by broofa** [comment]
Closing.  As it stands, this is an `@types/uuid` issue.   If/when we integrate TS support here, this should get fixed so I consider this a dupe of #654.
